hare/preroundtracker.go

```diff
@@ -29,20 +29,24 @@ func (pre *PreRoundTracker) OnPreRound(msg *pb.HareMessage) {
 		return
 	}
 
-	// only handle first pre-round msg
-	if _, exist := pre.preRound[verifier.String()]; exist {
+	var sToTrack *Set = nil
```

Comment:
you can refactor the else using better arrangement of parameters, sToTrack can start with msg.Message.Values as initial value and be modified else

Reply:
I think it is a matter of style but do understand it is a more simple and go-idiomatic approach so I changed it to fit the no else and use if for non-happy flows